### DISCUSSION TITLE
Auto-embed untranslated phrases

### DISCUSSION THREAD

**Discussion by mohamedsabil83** [Laravel-Lang/common#55] [Q&A]
I suggest auto-embed untranslated phrases in all languages files. It's easier to translate like that instead of checking the place of the phrase in the en files then paste and translate in the same place in the target language(s).

**Comment by olsza** [comment]
it's a good idea, but I will think about an optional solution, i.e. call a script with a parameter, where the parameter would be a language abbreviation, where it should automatically add the missing words

**Comment by andrey-helldar** [comment]
The idea is very good, but the problem is that PHP does not export variables by default:
1. in a "beautiful" view;
2. with comments.

That is, the problem will be that you will need to install a dependency for formatting arrays, for example, [this one](https://github.com/andrey-helldar/pretty-array), and refuse to store comments in them, for example, [these](https://github.com/Laravel-Lang/lang/blob/master/src/ar/validation.php#L4-L13) or [these](https://github.com/Laravel-Lang/lang/blob/master/src/ar/validation.php#L120-L129).

In my opinion, comments are an important part of files, as they tell people in their language what they contain.

**Reply by olsza** [reply]
you're right @andrey-helldar , comments are important and even I didn't think to delete them;)

but maybe before we get to this idea, we will need to refactor the code, among others break each class in todo.php into separate files to make corrections easier.

ps. In fact, in the form that is the code now, you can do this improvement, but I think that refreshing the code would be useful;)

**Comment by olsza** [comment]
maybe, for now, do auto-add in JSON, because that's where the most is currently happening, and the rest in the future, as the team thinks?
because it is a faster action, and it can save others a lot of time ;)

**Reply by andrey-helldar** [reply]
I like it

**Comment by olsza** [comment]
[![](https://api.gh-polls.com/poll/01EM66P7Z4RDKVYDE5G07H2D6N/auto%20add%20in%20all%20files%20%5BJSON%20and%20array%20in%20.php%5D)](https://api.gh-polls.com/poll/01EM66P7Z4RDKVYDE5G07H2D6N/auto%20add%20in%20all%20files%20%5BJSON%20and%20array%20in%20.php%5D/vote)
[![](https://api.gh-polls.com/poll/01EM66P7Z4RDKVYDE5G07H2D6N/auto-add%20first%20in%20JSON)](https://api.gh-polls.com/poll/01EM66P7Z4RDKVYDE5G07H2D6N/auto-add%20first%20in%20JSON/vote)
[![](https://api.gh-polls.com/poll/01EM66P7Z4RDKVYDE5G07H2D6N/%20we%20don't%20add%20auto-add)](https://api.gh-polls.com/poll/01EM66P7Z4RDKVYDE5G07H2D6N/%20we%20don't%20add%20auto-add/vote)

**Reply by olsza** [reply]
I was wondering why you can't see the poll? but now I know ... because the domain api.gh-polls.com is not up to date certificate: /

**Comment by olsza** [comment]
Ok, I can see very the big majority for auto-add first in JSON.
It will take up for it :)
The question is, should I place Ticket on Issues?

**Reply by andrey-helldar** [reply]
I think, PR is enough.
